### PR TITLE
Add support for PSR-17 HTTP factories

### PR DESCRIPTION
**Pull request by Nyholm**
This will add support for PSR-17. 

While waiting for this to be merged, one can use this package: https://github.com/http-interop/http-factory-guzzle

#### TODO

* [x] Update the tests

#### Blockers

* [ ] This requires us to drop PHP5 support
* [x] We also need to wait for the official PSR package.

### CODE REVIEW

**Comment by Nyholm** [comment]
This PR is ready for a review. That only blocker is to drop PHP5 support.
We could do that in a minor release.

**Comment by Nyholm** [comment]
Okey. That is fair. I added this to the 2.0 milestone.

File: src/Factory/HttpFactory.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    {
+        return \GuzzleHttp\Psr7\stream_for($resource);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createServerRequest(string $method, $uri, array $serverParams = []): ServerRequestInterface
+    {
+        if (empty($method)) {
```
Comment (gmponos) [inline src/Factory/HttpFactory.php]: How would you feel about using `$method === ''`

File: src/Factory/HttpFactory.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+use Psr\Http\Message\UploadedFileFactoryInterface;
+use Psr\Http\Message\UriFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UploadedFileInterface;
+use Psr\Http\Message\UriInterface;
+
+class HttpFactory implements
```
Comment (gmponos) [inline src/Factory/HttpFactory.php]: just a thought here. Should this be final or not?

File: src/Factory/HttpFactory.php (hunk 1 above)
Comment (Nyholm) [reply src/Factory/HttpFactory.php]: Could you elaborate?

Comment (gmponos) [reply src/Factory/HttpFactory.php]:
```
$val = '0';
$test = empty($val);
var_dump($test); // true
```

There are a lot of libraries for Codestyle and codequality that they suggest avoid using empty.

File: composer.json [hunk 3]
```diff
@@ -16,12 +16,14 @@
...
     "require": {
-        "php": ">=5.4.0",
+        "php": "^7.0",
         "psr/http-message": "~1.0",
+        "psr/http-factory": "^1.0",
         "ralouphie/getallheaders": "^2.0.5"
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.8.36 || ^5.7.27 || ^6.5.8"
+        "phpunit/phpunit": "^6.5.8",
```
Comment (gmponos) [inline composer.json]: maybe we could try using 7.0 also.

**Comment by sagikazarmark** [comment]
Fixed in #238

**Pull request closed without merging**